sdk/test-utils/recorder/README.md

```diff
@@ -1,9 +1,450 @@
+      env.AZURE_TENANT_ID,
+      env.AZURE_CLIENT_ID,
+      env.AZURE_CLIENT_SECRET
+    );
+
+    const keyVaultUrl = "https://myKeyVault.vault.azure.net";
+    client = new KeysClient(keyVaultUrl, credential);
+  });
+
+  afterEach(function () {
+    recorder.stop();
+  });
+});
+```
+
+If you run the previous test with Mocha, and you set the `TEST_MODE`
+environment variable to `record`, the common recorder will create a recording file located in
+`recordings/node/my_test/recording_before_each_hook.js` with the contents of the HTTP request
+as well as the contents of the HTTP response.
+
+You'll see in the code above that we're calling to `recorder.stop`. This is so
+that, after each test, we can stop recording and the test file can be
+generated. We recommend creating new recorders on `beforeEach` and stopping the
+recorder on `afterEach` to make sure that the generated files are smaller and
+easier to understand than by having them all in one chunk.
+
+> **Note:** By this point you might have noticed that the values in the environment variables will be stored in the recordings.
+> We'll make sure this doesn't happen in the [Securing sensitive data](#securing-sensitive-data) example.
+
+To add recorded tests, feel free to extend this file with as many subsequent
+`it()` calls as you need, each one of them is going to end up in their appropriate file.
+
+```typescript
+  beforeEach(async function() {
+    recorder = record(that);
+    const credential = await new ClientSecretCredential(
+      env.AZURE_TENANT_ID,
+      env.AZURE_CLIENT_ID,
+      env.AZURE_CLIENT_SECRET
+    );
+
+    const keyVaultUrl = "https://myKeyVault.vault.azure.net";
+    client = new KeysClient(keyVaultUrl, credential);
+  });
+
+  afterEach(function () {
+    recorder.stop();
+  });
+
+  it("my first test", async function() {
+    const result = await client.createKey("First key", "RSA");
+    assert.equal(result.name, "First key");
+  });
+
+  it("my second test", async function() {
+    const result = await client.createKey("Second key", "RSA");
+    assert.equal(result.name, "Second key");
+  });
+
+  // And so on...
+});
+```
+
+### How to playback
+
+Once you have recorded something, you can run your tests again with `TEST_MODE`
+set to `playback`.  You'll notice how they pass much faster. This time, they
+will not reach out to the remote address, but instead they will respond every
+request according to their matching copy stored in the recordings.
+
+### Update existing recordings
+
+Once you have your recorded files, to update them after changing one of the tests, simply
+re-run the tests with `TEST_MODE` set to `record`. This will overwrite previously existing files.
+
+> **Note:** If you rename the file of the test, or the name of the test, the path
+> of the recording will change. If you do these kind of changes, make sure to delete your recordings folder.
+> The common recorder will re-generate only the updated tests.
```

Comment:
Possibly future request would be to detect this. If we have stale data from a test that's no longer available, we could do this step for them.

Reply:
Definitely! We're contemplating this in the improvements epic: https://github.com/Azure/azure-sdk-for-js/issues/4229 , specifically in issues like:

- https://github.com/Azure/azure-sdk-for-js/issues/4879
- https://github.com/Azure/azure-sdk-for-js/issues/5156